src/v/cluster/controller_snapshot.cc

```diff
@@ -11,6 +11,28 @@
 
 #include "cluster/controller_snapshot.h"
 
+#include "security/types.h"
+#include "serde/rw/rw.h"
+
+namespace serde {
+void tag_invoke(
+  tag_t<write_tag>,
+  iobuf& out,
+  cluster::controller_snapshot_parts::security_t::named_role t) {
+    write(out, std::move(t.name));
```

Comment:
@BenPope @oleiman Skipping the envelope saves only 2 bytes (the version and compact version are both int8_t). For that savings you have to write more code, and you can never change the encoding. I agree it's unlikely it will change, but why risk it?

Reply:
https://github.com/redpanda-data/redpanda/pull/17621